Objects/dictobject.c

```diff
@@ -3670,10 +3800,16 @@ dictiter_reduce(dictiterobject *di, PyObject *Py_UNUSED(ignored))
         PyObject *element = 0;
         if (Py_TYPE(di) == &PyDictIterItem_Type)
             element = dictiter_iternextitem(&tmp);
+        else if (Py_TYPE(di) == &PyDictRevIterItem_Type)
+            element = dictreviter_iternext(&tmp);
         else if (Py_TYPE(di) == &PyDictIterKey_Type)
             element = dictiter_iternextkey(&tmp);
+        else if (Py_TYPE(di) == &PyDictRevIterKey_Type)
+            element = dictreviter_iternext(&tmp);
         else if (Py_TYPE(di) == &PyDictIterValue_Type)
             element = dictiter_iternextvalue(&tmp);
+        else if (Py_TYPE(di) == &PyDictRevIterValue_Type)
+            element = dictreviter_iternext(&tmp);
         else
             Py_UNREACHABLE();
```

Comment:
Can we use switch-case here?

Reply:
I'm not sure it's possible don't `case`s needs to be int consts?

I tried and got this output from the compiler: `error: expression is not an integer constant expression`.

Reply:
You're right.